gmapping/src/replay.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2015 Aldebaran
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <ros/ros.h>
+
+#include "slam_gmapping.h"
+#include "boost/program_options.hpp"
```

Comment:
If I may, the order should be:

    #include "slam_gmapping.h"
    
    #include <iostream>
    #include <string>
    
    #include <boost/program_options.hpp>
    #include <ros/ros.h>

Because if your "slam_gmapping.h" header does not include the proper headers, you might not see the problem if you always include system headers first. Always include your lib headers first.